src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -31,6 +45,12 @@ public static ApplicationModel.IResourceBuilder<T> WithBuildArg<T>(this Applicat
         public static ApplicationModel.IResourceBuilder<T> WithBuildSecret<T>(this ApplicationModel.IResourceBuilder<T> builder, string name, ApplicationModel.IResourceBuilder<ApplicationModel.ParameterResource> value)
             where T : ApplicationModel.ContainerResource { throw null; }
 
+        public static ApplicationModel.IResourceBuilder<T> WithContainerFiles<T>(this ApplicationModel.IResourceBuilder<T> builder, string destinationPath, System.Collections.Generic.IEnumerable<ApplicationModel.ContainerFileSystemItem> entries, int defaultOwner = 0, int defaultGroup = 0, System.IO.UnixFileMode? umask = null)
```

Comment:
@danegsta @karolz-ms - do we want to hard-code `0` in every call site for these `WithContainerFiles` methods? Would it be better if `int? defaultOwner = null, int? defaultGroup = null`, so we could have the possibility of changing the default, if we ever needed to?

Reply:
I think this makes sense... on the DCP side we assume 0/0 if the request does not carry the default. What do you think @danegsta ?

Reply:
Yeah, makes sense to future proof with null as the default.

Reply:
Opened https://github.com/dotnet/aspire/pull/8557 (and I'll open a backport to 9.2 if that looks good)